### ISSUE TITLE
Geo IP with online queries

### ISSUE THREAD

**Issue by network-scripts** [OpenBMP/obmp-grafana#4]
I have modified the Prefix Location panel in the Looking Glass dashboard to query the GeoIP location from RIPE instead of database. It needs the yesoreyeram-infinity-datasource grafana plugin. I am not sure, this is to be implemented in the live version, since everything in there tends to be offline. Someone might find this usefull though, so I am posting it here, where everyone will most likely look :).

```json
  {
      "circleMaxSize": "3",
      "circleMinSize": "1",
      "colors": [
        "#73BF69",
        "#FF9830",
        "#F2495C"
      ],
      "datasource": {
        "type": "yesoreyeram-infinity-datasource",
        "uid": "pSm4O7r7z"
      },
      "decimals": 1,
      "esMetric": "Count",
      "gridPos": {
        "h": 8,
        "w": 12,
        "x": 12,
        "y": 0
      },
      "hideEmpty": false,
      "hideZero": false,
      "id": 17,
      "initialZoom": "1.5",
      "locationData": "table",
      "mapCenter": "(0°, 0°)",
      "mapCenterLatitude": 0,
      "mapCenterLongitude": 0,
      "maxDataPoints": 1,
      "mouseWheelZoom": false,
      "showLegend": false,
      "stickyLabels": false,
      "tableQueryOptions": {
        "geohashField": "geohash",
        "labelField": "name",
        "latitudeField": "latitude",
        "longitudeField": "longitude",
        "metricField": "value",
        "queryType": "coordinates"
      },
      "targets": [
        {
          "columns": [
            {
              "selector": "latitude",
              "text": "latitude",
              "type": "string"
            },
            {
              "selector": "longitude",
              "text": "longitude",
              "type": "string"
            },
            {
              "selector": "country",
              "text": "name",
              "type": "string"
            },
            {
              "selector": "covered_percentage",
              "text": "value",
              "type": "string"
            }
          ],
          "datasource": {
            "type": "yesoreyeram-infinity-datasource",
            "uid": "pSm4O7r7z"
          },
          "filters": [],
          "format": "table",
          "global_query_id": "",
          "refId": "A",
          "root_selector": "$.*.*.*.locations",
          "source": "url",
          "type": "json",
          "url": "http://stat.ripe.net/data/maxmind-geo-lite/data.json?resource=$input",
          "url_options": {
            "data": "",
            "method": "GET"
          }
        }
      ],
      "thresholds": "2,99",
      "title": "Prefix Location",
      "type": "grafana-worldmap-panel",
      "unitPlural": "%",
      "unitSingle": "",
      "unitSingular": "%",
      "valueName": "current"
    },
```

**Comment by TimEvens** [comment]
Isn't RIPE just using maxmind geo-lite? The DB doesn't auto load maxmind or dbip lite by default.  There is a script to load DBIP.  Maybe what's needed is to automate the load based on selection (maxmind geo-lite or DBIP lite)...  The reason we do want to use the DB for many geo lookups is for faster queries, offline support, and the ability to control/update geo as you need. For example, you can add geo-coding to any internal prefix, l3vpn, ...

**Comment by network-scripts** [comment]
That is what I figured.  I have found the `dbip-to-psql` script only now, when you mentioned it. Before I was not able to find any clue on how to easily fill the database, without writing my own parser. I have chosen RIPE just because they do not need any registration to use the api. It just works out of the box. Anyhow, I wrote the issue just so anyone else wondering on how to get the world map panel to work, there is a script in the `psql-app` container. Or there is my way :). Thank you for pointing me in the right direction.
